File: android/BaseAndroidXMLKtor/app/src/main/java/dev/subfly/baseandroidxmlktor/featureHome/view/HomeState.kt
package dev.subfly.baseandroidxmlktor.featureHome.view

import dev.subfly.baseandroidxmlktor.core.util.Constants
import dev.subfly.baseandroidxmlktor.core.domain.model.CoinHomeModel

data class HomeState(
    val coins: List<CoinHomeModel> = emptyList(),
    val isLoading: Boolean = true,
    val error: String = Constants.EMPTY_STRING,
)

File: ios/BaseiOSUIKit/BaseiOSUIKit/FeatureHome/View/HomeState.swift
//
//  HomeState.swift
//  BaseiOSUIKit
//
//  Created by Ali Taha on 5.12.2023.
//

import Foundation

struct HomeState {
    var isLoading: Bool = true
    var error: String = ""
    var coins: [CoinHomeModel] = []
    
    func copy(
        isLoading: Bool? = nil,
        error: String? = nil,
        coins: [CoinHomeModel]? = nil
    ) -> HomeState {
        return HomeState(
            isLoading: isLoading ?? self.isLoading,
            error: error ?? self.error,
            coins: coins ?? self.coins
        )
    }
}
